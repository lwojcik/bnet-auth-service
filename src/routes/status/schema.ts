const response = {
  200: {
    type: "object",
    properties: {
      status: { type: "number" },
      message: { type: "string" },
      timestamp: { type: "string" },
    },
  },
  400: {
    type: "object",
    properties: {
      status: { type: "number" },
    },
  },
  500: {
    type: "object",
    properties: {
      status: { type: "number" },
    },
  },
};

const schema = {
  response,
};

export default schema;
