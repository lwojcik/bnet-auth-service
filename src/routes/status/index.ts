import { FastifyPluginCallback } from "fastify";
import fp from "fastify-plugin";
import schema from "./schema";

// eslint-disable-next-line no-empty-pattern
const route: FastifyPluginCallback<Record<string, never>> = (
  server,
  // eslint-disable-next-line no-empty-pattern
  {},
  next
) => {
  // eslint-disable-next-line no-empty-pattern
  server.get("/status", { schema }, ({}, reply) => {
    reply.code(200).send({
      status: 200,
      message: "ok",
      timestamp: new Date(),
    });
  });
  next();
};

export default fp(route);
