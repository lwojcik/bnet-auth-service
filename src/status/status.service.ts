import { Injectable } from '@nestjs/common';
import { Duration } from 'luxon';
import { StatusResponse } from './dto/status-response.dto';

@Injectable()
export class StatusService {
  getStatus(): StatusResponse {
    const uptimeSeconds = Math.floor(process.uptime());
    const uptimeMinutes = Math.floor(uptimeSeconds / 60);
    const uptimeHours = Math.floor(uptimeMinutes / 60);
    const uptimeDays = Math.floor(uptimeHours / 24);

    return {
      status: 'ok',
      uptime: Duration.fromObject({
        days: uptimeDays,
        hours: uptimeHours,
        minutes: uptimeMinutes,
      }).toHuman({
        unitDisplay: 'narrow',
      }),
      timestamp: new Date().toISOString(),
    };
  }
}
