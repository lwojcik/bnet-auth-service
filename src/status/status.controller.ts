import { Controller, Get } from '@nestjs/common';
import { StatusService } from './status.service';

@Controller('status')
export class StatusController {
  constructor(private readonly statusService: StatusService) {}

  @Get()
  getStatus() {
    return this.statusService.getStatus();
  }
}
