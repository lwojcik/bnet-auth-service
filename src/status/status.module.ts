import { Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import { authConfig } from '../config';
import { LoggerModule } from '../logger/logger.module';
import { StatusController } from './status.controller';
import { StatusService } from './status.service';

@Module({
  imports: [LoggerModule, ConfigModule.forFeature(authConfig)],
  controllers: [StatusController],
  providers: [StatusService],
})
export class StatusModule {}
