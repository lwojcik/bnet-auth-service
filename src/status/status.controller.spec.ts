import { Test, TestingModule } from '@nestjs/testing';
import { StatusController } from './status.controller';

describe('StatusController', () => {
  let controller: StatusController;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      controllers: [StatusController],
    }).compile();

    controller = module.get<StatusController>(StatusController);
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
  });
});
