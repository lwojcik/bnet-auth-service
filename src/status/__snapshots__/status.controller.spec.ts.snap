// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusController should get status 1`] = `
{
  "testStatusProperty": "testStatusValue",
}
`;
